packages/desktop/src/services/FileDownloadServiceElectron.ts

```diff
@@ -317,12 +419,92 @@ export default class FileDownloadServiceElectron
         });
     }
 
-    public async prepareHttpResourceForDownload(url: string, postBody: string): Promise<Blob> {
-        const response = await this.rawPost<string>(url, postBody);
-        return new Blob([response], { type: "application/json" });
+    private async downloadS3Directory(
+        fileInfo: FileInfo,
+        downloadRequestId: string,
+        onProgress?: (transferredBytes: number) => void,
+        destination?: string
+    ): Promise<DownloadResult> {
+        const { bucket, key, region } = this.parseS3Url(fileInfo.path);
+        const destinationDir = destination || (await this.getDefaultDownloadDirectory());
+        const fullDestinationDir = path.join(destinationDir, fileInfo.name);
+
+        try {
+            // Ensure the destination directory exists
+            fs.mkdirSync(fullDestinationDir, { recursive: true });
+
+            const keys = await this.listS3Objects(bucket, key, region);
+
+            if (keys.length === 0) {
+                throw new Error("No files found in the specified S3 directory.");
+            }
+
+            for (const fileKey of keys) {
+                if (!fileKey) {
+                    console.warn(`Encountered null or undefined file key. Skipping.`);
+                    continue;
+                }
+
+                const relativePath = path.relative(key, fileKey);
+                const destinationPath = path.join(fullDestinationDir, relativePath);
+
+                // Ensure the subdirectories exist
+                fs.mkdirSync(path.dirname(destinationPath), { recursive: true });
+
+                const fileUrl = `https://${bucket}.s3.${region}.amazonaws.com/${encodeURIComponent(
```

Comment:
Don't we have this initial host name from the file's "File Path"?

Reply:
oh whoops, a symptom of trying lots of bits and pieces. will fix

Reply:
resolved in [cc3d7a5](https://github.com/AllenInstitute/biofile-finder/pull/204/commits/cc3d7a5d5be38541d1c9db2c770d19809f60a919)